src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/MsQuicAddressHelpers.cs

```diff
@@ -3,17 +3,18 @@
 
 using System.Diagnostics;
 using System.Net.Sockets;
+using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
 using Microsoft.Quic;
 
 namespace System.Net.Quic.Implementations.MsQuic.Internal
 {
     internal static class MsQuicAddressHelpers
     {
-        internal static unsafe IPEndPoint INetToIPEndPoint(IntPtr pInetAddress)
+        internal static unsafe IPEndPoint ToIPEndPoint(this ref QuicAddr quicAddress)
         {
             // MsQuic always uses storage size as if IPv6 was used
-            Span<byte> addressBytes = new Span<byte>((byte*)pInetAddress, Internals.SocketAddress.IPv6AddressSize);
+            Span<byte> addressBytes = new Span<byte>((byte*)Unsafe.AsPointer(ref quicAddress), Internals.SocketAddress.IPv6AddressSize);
```

Comment:
This is potential corruption if quicAddress isn't fixed in memory.  I assume it is, in which case it's worth a comment.

Reply:
It's native memory, so it cannot be moved by GC. I'll add a comment.